src/main/scala/io/smartdatalake/workflow/dataobject/TickTockHiveTableDataObject.scala

```diff
@@ -55,7 +55,40 @@ case class TickTockHiveTableDataObject(override val id: DataObjectId,
   private val connection = connectionId.map(c => getConnection[HiveTableConnection](c))
 
   // prepare final path and table
-  @transient private[workflow] lazy val hadoopPath = HdfsUtil.prefixHadoopPath(path.getOrElse("."), connection.map(_.pathPrefix))
+  @transient private var hadoopPathHolder: Path = _
+  def hadoopPath(implicit session: SparkSession): Path = {
+    require(isTableExisting || path.isDefined, s"TickTockHiveTable ${table.fullName} does not exist, so path must be set.")
+
+    if (hadoopPathHolder == null) {
+      hadoopPathHolder = {
+        if (isTableExisting) new Path(HiveUtil.existingTableLocation(table))
+        else HdfsUtil.prefixHadoopPath(path.get, connection.map(_.pathPrefix))
+      }
+
+      // For existing tables, check to see if we write to the same directory. If not, issue a warning.
+      if(isTableExisting && path.isDefined) {
+        // Normalize both paths before comparing them (remove tick / tock folder and trailing slash)
+        val hadoopPathNormalized = hadoopPathHolder.toString
```

Comment:
Could we create a function HdfsUtils.normalizePath to avoid this code dupliation?

Reply:
That of course, makes a lot of sense: https://github.com/smart-data-lake/smart-data-lake/pull/126/commits/09f39d7ed596f34fd1ba6091043eeba698aa7055